### PR TITLE
[AARCH64][SVE] Add intrinsics for SVE LUTI instructions

### PR DESCRIPTION
**Pull request by Lukacma**
This patch adds intrinsics for LUTI2 and LUTI4 instructions, which use SVE registers, as specified in the https://github.com/ARM-software/acle/pull/324

### CODE REVIEW

File: llvm/lib/Target/AArch64/SVEInstrFormats.td [hunk 1]
```diff
@@ -10349,6 +10349,16 @@ multiclass sve2_luti2_vector_index<string mnemonic> {
     let Inst{23-22} = idx{2-1};
     let Inst{12}    = idx{0};
   }
+
```
Comment (CarolineConcatto) [inline llvm/lib/Target/AArch64/SVEInstrFormats.td:10352]: I can see lots of white spaces  when I push the patch at the end of:
def : SVE_3_Op_Imm_Pat
and
def : Pat<(nxv8i16 (int_aarch64_sve_luti4_lane_x2

File: clang/test/CodeGen/aarch64-sve2-intrinsics/acle_sve2_luti.c [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+// REQUIRES: aarch64-registered-target
+// RUN: %clang_cc1 -fclang-abi-compat=latest -triple aarch64-none-linux-gnu \
+// RUN:   -target-feature +sme -target-feature +sme2 -target-feature +lut -target-feature +bf16 -O1 -Werror -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fclang-abi-compat=latest -DSVE_OVERLOADED_FORMS -triple aarch64-none-linux-gnu \
+// RUN:   -target-feature +sve -target-feature +sve2 -target-feature +lut -target-feature +bf16 -O1 -Werror -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fclang-abi-compat=latest -triple aarch64-none-linux-gnu \
+// RUN:   -target-feature +sve -target-feature +sve2 -target-feature +lut -target-feature +bf16 -O1 -Werror -emit-llvm -o - -x c++ %s | FileCheck %s -check-prefix=CPP-CHECK
+// RUN: %clang_cc1 -fclang-abi-compat=latest -DSVE_OVERLOADED_FORMS -triple aarch64-none-linux-gnu \
+// RUN:   -target-feature +sve -target-feature +sve2 -target-feature +lut -target-feature +bf16 -O1 -Werror -emit-llvm -o - -x c++ %s | FileCheck %s -check-prefix=CPP-CHECK
+#include <arm_sve.h>
```
Comment (CarolineConcatto) [inline clang/test/CodeGen/aarch64-sve2-intrinsics/acle_sve2_luti.c:11]: Can you add this run line
// RUN: %clang_cc1 -triple aarch64 -target-feature +sme -target-feature +sme2 -target-feature +lut -target-feature +bf16 -S -O1-Werror -Wall -o /dev/null %s$

Comment (Lukacma) [reply clang/test/CodeGen/aarch64-sve2-intrinsics/acle_sve2_luti.c:12]: doen

File: llvm/lib/Target/AArch64/SVEInstrFormats.td (hunk 1 above)
Comment (Lukacma) [reply llvm/lib/Target/AArch64/SVEInstrFormats.td:10352]: I am not sure I understood this, but I removed extra whitespaces

Comment (Lukacma) [reply llvm/lib/Target/AArch64/SVEInstrFormats.td:10352]: Done

**Review by CarolineConcatto — APPROVED** [review]
LGTM

**Pull request merged**